contracts/strategies/AlphaHomoraStrategy.sol

```diff
@@ -0,0 +1,321 @@
+// SPDX-License-Identifier: agpl-3.0
+pragma solidity ^0.8.0;
+
+/**
+ * @title XUSD AlphaHomora Strategy
+ * @notice Investment strategy for investing stablecoins via AlphaHomora/CREAM
+ * @author XUSD.fi Inc
+ */
+import { SafeERC20 } from "@openzeppelin/contracts/token/ERC20/utils/SafeERC20.sol";
+
+import { IERC20, InitializableAbstractStrategy } from "../utils/InitializableAbstractStrategy.sol";
+import { IVault } from "../interfaces/IVault.sol";
+import { ICERC20 } from "../interfaces/alphaHomora/ICERC20.sol";
+import { ISafeBox } from "../interfaces/alphaHomora/ISafeBox.sol";
+import { IAlphaIncentiveDistributor } from "../interfaces/alphaHomora/IAlphaIncentiveDistributor.sol";
+
+contract AlphaHomoraStrategy is InitializableAbstractStrategy {
+    using SafeERC20 for IERC20;
+
+    address[] public incentiveControllerAddresses;
+    mapping(address => bytes32[]) internal _proofs;
+    mapping(address => uint256) internal _amounts;
+
+    function initialize(
+        address _platformAddress, // dead
+        address _vaultAddress,
+        address[] calldata _rewardTokenAddresses, // [ALPHA, WAVAX]
+        address[] calldata _assets,
+        address[] calldata _pTokens,
+        address[] calldata _incentiveControllerAddresses // [ALPHAcontrollerAddr, WAVAXcontrollerAddr]
+    ) external onlyGovernor initializer {
+        require(
+            _rewardTokenAddresses.length ==
+                _incentiveControllerAddresses.length,
+            "not 1:1 rewards-to-incentives"
+        );
+        incentiveControllerAddresses = _incentiveControllerAddresses;
+
+        InitializableAbstractStrategy._initialize(
+            _platformAddress,
+            _vaultAddress,
+            _rewardTokenAddresses,
+            _assets,
+            _pTokens
+        );
+    }
+
+    event SkippedWithdrawal(address asset, uint256 amount);
+
+    /**
+     * @dev Collect accumulated WAVAX+ALPHA and send to Vault.
+     */
+    function collectRewardTokens() external override onlyVault nonReentrant {
+        for (uint256 i = 0; i < rewardTokenAddresses.length; i++) {
+            IAlphaIncentiveDistributor _incentivesController = IAlphaIncentiveDistributor(
+                    incentiveControllerAddresses[i]
+                );
+            require(_incentivesController.token() == rewardTokenAddresses[i]);
+            uint256 _amount = _amounts[rewardTokenAddresses[i]];
+            bytes32[] memory _proof = _proofs[rewardTokenAddresses[i]];
+            uint256 _claimed = _incentivesController.claimed(address(this));
```

Comment:
will _claimed be 0 for first reward period, or does this throw if this is not in claimed?